### PR TITLE
Add 'Close' & 'Back' aria-label attributes to buttons.

### PR DESCRIPTION
**Pull request by grimmdude**
**Changes proposed in this pull request:**
Adds `aria-label` attribute to modal close & back buttons to improve accessibility and satisfy these warnings.

Screen Shot 2021-11-11 at 10 08 32 AM

**Reviewers should focus on:**
Affects every modal and the hero welcome component.

**Necessity**

- [x] Has the problem that is being solved here been clearly explained?
- [x] If applicable, have various options for solving this problem been considered?
- [x] For core PRs, does this need to be in core, or could it be in an extension?
- [x] Are we willing to maintain this for years / potentially forever?

**Confirmed**

- [x] Frontend changes: tested on a local Flarum installation.
- [x] Backend changes: tests are green (run `composer test`).
- [x] Core developer confirmed locally this works as intended.
- [x] Tests have been added, or are not appropriate here.

### CODE REVIEW

File: js/src/common/components/Navigation.js [hunk 1]
```diff
@@ -53,6 +53,7 @@ export default class Navigation extends Component {
         e.preventDefault();
         history.back();
       },
+      'aria-label': app.translator.trans('core.lib.close_button'),
```
Comment (davwheat) [inline js/src/common/components/Navigation.js]: I think it would be more appropriate to have it as "Back" rather than "Close".

Comment (grimmdude) [reply js/src/common/components/Navigation.js]: Agreed @davwheat , my mistake.  Just pushed that change.

File: js/src/forum/components/WelcomeHero.js [hunk 2]
```diff
@@ -27,6 +27,7 @@ export default class WelcomeHero extends Component {
             icon: 'fas fa-times',
             onclick: slideUp,
             className: 'Hero-close Button Button--icon Button--link',
+            'aria-label': app.translator.trans('core.lib.close_button'),
```
Comment (davwheat) [inline js/src/forum/components/WelcomeHero.js:30]: I'd be a fan of a more descriptive accessible label here, like "Hide welcome message". What do you think?

File: locale/core.yml [hunk 3]
```diff
@@ -500,6 +500,7 @@ core:
   # Translations in this namespace are used by the forum and admin interfaces.
   lib:
     debug_button: Debug
+    close_button: => core.ref.close
```
Comment (davwheat) [inline locale/core.yml:503]: I'd prefer this to be listed under a `modal` key to ensure we don't use this in other components too. We try to keep to separate translation keys for each instance of text.

File: js/src/common/components/Navigation.js [hunk 4]
```diff
@@ -53,6 +53,7 @@ export default class Navigation extends Component {
         e.preventDefault();
         history.back();
       },
+      'aria-label': app.translator.trans('core.lib.nav.back'),
```
Comment (SychO9) [inline js/src/common/components/Navigation.js:56]: It'd be better to replace the title parameter above with `aria-label` instead, `previous.title` is already being implicitly used right now as `aria-label` and is more descriptive.

**Review by SychO9 — COMMENTED** [review]
Thank you for the PR!

**Review by SychO9 — APPROVED** [review]
Looks good to me.

**Review by askvortsov1 — APPROVED** [review]
Thank you!

**Pull request merged**